### PR TITLE
Stop using singletons, isEnabled, flags via config, cust service names

### PR DESCRIPTION
**Pull request by mixonic**
A big batch of work! It got away from me a bit :-/

* Stop using a singleton service. This makes testing far easier,
  removing the need for a reset.
* Use `isEnabled` and deprecate `enabled`. This is to distinguishes `enabled`
  from `enable`, which seemed a perilous combo.
* Allow flags to be set via `featureFlags` in the config (not on app)
* Deprecate setting config via `APP.FEATURES`. Anti-pattern.
* Add the ability to customize service names via `featureFlagService`
* Flesh out tests
* Split the abstract features class from the self-configuring one
* Stop injecting onto view, helper since they are not reliable for
  injections
* Make the flags map private and a fast object
* Make `featureIsEnabled` private
* Change acceptance helpers to take the app as an argument
* Improve tests

The [updated README can be read here](https://github.com/mixonic/ember-feature-flags/blob/configurable/README.md)

### CODE REVIEW

File: addon/tests/helpers/reset-feature-flags.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import Ember from 'ember';
-import features from 'ember-feature-flags/features';
 
-export function resetFeatureFlags(){
-  features.setup({});
+export function resetFeatureFlags (app) {
```
Comment (kategengler) [inline addon/tests/helpers/reset-feature-flags.js]: Should this stick around if its no longer needed?

File: README.md [hunk 2]
```diff
@@ -3,102 +3,132 @@
...
-      if (this.features.enabled('new-billing-plans')){
-         // Return new plans
-      } else {
-         // Return old plans
-      }
-    }.property()
-  });
+export default Ember.Controller.extend({
+  plans: function(){
+    if (this.features.enabled('new-billing-plans')){
```
Comment (kategengler) [inline README.md]: `enabled` is now `isEnabled`, right?

File: tests/unit/features/base-test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import {
```
Comment (kategengler) [inline tests/unit/features/base-test.js]: Duplicate test file with test/unit/features/main-test.js ?

**Comment by kategengler** [comment]
Had a few comments but other than that it's awesome. Thanks!

File: addon/tests/helpers/reset-feature-flags.js (hunk 1 above)
Comment (mixonic) [inline addon/tests/helpers/reset-feature-flags.js]: Uncertain. I think the right thing to do may be to deprecate it, I'll do that.

**Comment by mixonic** [comment]
@kategengler addressed a few items, let me know if you have other thoughts. Eager to add this to an app tomorrow :-)

**Pull request merged**